Add SCIM provisioning documentation for HCP (#2561)
hashicorp/web-unified-docs: content/hcp-docs/content/docs/hcp/iam/scim/troubleshooting.mdx

1. If you have reached a quota limit, contact [HashiCorp support](https://support.hashicorp.com/) to request a limit increase
1. If attribute mappings are incorrect, update them in your identity provider according to the [SCIM provisioning documentation](/hcp/docs/hcp/iam/sso/scim)
1. If network connectivity is an issue, verify that your identity provider can reach `https://api.cloud.hashicorp.com`

## Attribute mapping errors

Attribute mapping errors occur when required user or group attributes are missing or incorrectly configured in your identity provider.

### Common attribute issues

- **Missing required attributes** - The `userName`, `emails`, `name.givenName`, or `name.familyName` attributes are not mapped
- **Incorrect attribute format** - Attributes are mapped to the wrong SCIM field
- **Unsupported attributes** - Your identity provider is sending attributes that HCP does not support

### Provider-specific considerations

**Microsoft Entra ID:**
- Ensure you have removed unsupported default attribute mappings
- Verify that the email identifier is derived from the **Email** field under **Contact information**, not the **User principal name** field

> - Confirm that you have appended `?aadOptscim062020` to the Base URL

Reviewer: In the UI we do provide a Entra-specific base URL with that feature flag already appended. Would the language 

```- Confirm that `?aadOptscim062020` is appended to the Base URL``` 

be more clear?

Reviewer 2: ```suggestion
- HCP generates a base URL specific to Entra ID. Confirm that you have copied the correct base URL to your IdP. Refer to [Enable SCIM provisioning on HCP](/hcp/docs/hcp/iam/scim/entra-id#enable-scim-provisioning-on-hcp) for more information.
```

**Okta:**
- Verify that the API token includes the `Bearer ` prefix
- Ensure that **Create users**, **Update User Attributes**, and **Deactivate Users** are enabled

### How to verify correct mappings

1. Review the attribute mappings in your identity provider
1. Compare them to the attribute mapping tables in the [SCIM provisioning documentation](/hcp/docs/hcp/iam/sso/scim)
1. Ensure that all required attributes are mapped
1. Remove any unsupported attribute mappings

## Quota exceeded errors

HCP enforces quotas to ensure optimal performance. If you exceed a quota, SCIM sync fails for the affected resource type.

### Understanding quota limits

For current quota limits, refer to the [HCP support page](/hcp/docs/hcp/admin/support).

### How to check current usage

> 1. [Log in to HCP](https://portal.cloud.hashicorp.com/) and go to your organization

Reviewer: These steps are not correct. The counts are not on the SCIM page - they're on the user and group pages at the org level. The steps would be:

1. [Log in to HCP](https://portal.cloud.hashicorp.com/) and go to your organization
2. Click **Access control (IAM)**
3. Click **Users** to review the current count of users and **Groups**to review the current count of groups.

1. Click **Organization settings** and then **SCIM provisioning**

> 1. Review the current count of users and groups

Reviewer 2: ```suggestion
1. Click **Access control (IAM)**.
1. Click **Users** to review the current count of users and **Groups** to review the current count of groups.
```

### Contact support for limit increases

If your organization requires higher limits, contact [HashiCorp support](https://support.hashicorp.com/) with the following information:

- Your organization ID
- The current quota limit
- The requested quota limit
- Business justification for the increase

## User or group not syncing

If a specific user or group is not syncing from your identity provider to HCP, complete the following troubleshooting steps.

### Verify SCIM is enabled

1. [Log in to HCP](https://portal.cloud.hashicorp.com/) and go to your organization
1. Click **Organization settings** and then **SCIM provisioning**

> 1. Verify that SCIM provisioning is enabled and the status is **Active**

Reviewer: Just a note that there is no explicit status of "Active" on the page at this time.

Reviewer 2: ```suggestion
1. Verify that SCIM provisioning is enabled and the status is active.
```

### Check identity provider assignment

1. Sign in to your identity provider
1. Navigate to the HCP application
1. Verify that the user or group is assigned to the application
1. If the user or group is not assigned, assign them and wait for the next sync cycle

### Verify attribute mappings

1. Review the attribute mappings for the user or group in your identity provider
1. Ensure that all required attributes are present and correctly formatted
1. For users, verify that the email address is valid and unique

### Check SCIM provisioning status

1. In HCP, navigate to **Users** or **Groups**
1. Locate the user or group in the list
1. Check the **SCIM provisioning** column for the status

> 1. If the status is not **Syncing**, the user or group may not be managed by SCIM

Reviewer: ```suggestion
1. The status is **Synced ** when the IdP manages the user or group using SCIM. Any other status indicates that the user or group is managed in HCP, even if access to HCP is managed through the IdP. 
```
Better?

Reviewer 3: Could this be re-worded to avoid the double negative of "is not Synced" and "may not be managed"?

### How duplicates occur

Duplicates can occur in the following scenarios:

- A user or group exists in HCP before SCIM is enabled, but the email address or name does not match exactly
- A user or group is manually created in HCP after SCIM is enabled
- Multiple identity provider users have the same email address

### Resolution steps

1. Identify the duplicate users or groups in HCP
1. Determine which user or group should be the source of truth
1. If the identity provider should be the source of truth, remove the duplicate from HCP
1. If HCP should be the source of truth, remove the duplicate from your identity provider
1. Wait for the next sync cycle to complete

## Token authentication failures

Token authentication failures occur when the SCIM token is invalid, expired, or incorrectly configured in your identity provider.

### Token expiration

SCIM tokens do not expire automatically, but they can be regenerated in HCP. If you regenerate a token, you must update it in your identity provider.

### Token regeneration steps

1. [Log in to HCP](https://portal.cloud.hashicorp.com/) and go to your organization
1. Click **Organization settings** and then **SCIM provisioning**

> 1. Click **Regenerate SCIM token**

Reviewer: ```suggestion
1. Click **Generate SCIM token**
```

There is no regenerate button
